inference-engine/tests/functional/shared_test_classes/src/read_ir/read_ir.cpp

```diff
@@ -0,0 +1,125 @@
+            std::vector<float> a;
+            blob = FuncTestUtils::createAndFillBlobWithFloatArray(info.getTensorDesc(), roiIdxVector.data(), roiIdxVector.size());
+    } else if (isRoiOperation(node) && info.getTensorDesc().getDims().size() == 2) {
+        blob = make_blob_with_precision(info.getTensorDesc());
+        blob->allocate();
+        const auto& inputShape = cnnNetwork.getInputShapes().begin()->second;
+
+        if (ngraph::is_type<ngraph::op::v0::PSROIPooling>(node)) {
+            const auto& PSROIPoolNode = std::dynamic_pointer_cast<ngraph::op::v0::PSROIPooling>(node);
+            PSROIPoolingLayerTest::fillROITensor(blob->buffer(),
+                                                 blob->size() / 5,
+                                                 inputShape[0],
+                                                 inputShape[2],
+                                                 inputShape[3],
+                                                 PSROIPoolNode->get_group_size(),
+                                                 PSROIPoolNode->get_spatial_scale(),
+                                                 PSROIPoolNode->get_spatial_bins_x(),
+                                                 PSROIPoolNode->get_spatial_bins_y(),
+                                                 PSROIPoolNode->get_mode());
+        } else if (ngraph::is_type<ngraph::op::v0::ROIPooling>(node)) {
+            const auto& ROIPoolNode = std::dynamic_pointer_cast<ngraph::op::v0::ROIPooling>(node);
+            CommonTestUtils::fill_data_roi(blob->buffer(),
+                                           blob->size(),
+                                           ROIPoolNode->get_input_shape(0).front() - 1,
+                                           inputShape[2],
+                                           inputShape[3],
+                                           1.0f,
+                                           ROIPoolNode->get_method() == "max");
+        } else if (ngraph::is_type<ngraph::op::v3::ROIAlign>(node)) {
+            const auto &ROIAlignNode = std::dynamic_pointer_cast<ngraph::op::v3::ROIAlign>(node);
+            std::vector<float> blobData(ROIAlignNode->get_shape()[0] * 4);
+            ROIAlignLayerTest::fillCoordTensor(blobData,
+                                               inputShape[2],
+                                               inputShape[3],
+                                               ROIAlignNode->get_spatial_scale(),
+                                               ROIAlignNode->get_sampling_ratio(),
+                                               ROIAlignNode->get_pooled_h(),
+                                               ROIAlignNode->get_pooled_w());
+            blob = FuncTestUtils::createAndFillBlobWithFloatArray(info.getTensorDesc(), blobData.data(), blobData.size());
+        }
+    } else {
+        blob = GenerateInput(info);
+    }
+    return blob;
+}
+
+void ReadIRTest::Infer() {
```

Comment:
Along with changes in https://github.com/openvinotoolkit/openvino/pull/4117 this Infer method overloading become redundant

Reply:
It's open. How will you offer to use this functionality without merge? I will use it after merge

Reply:
Done after merge